src/tbx/mod.rs

```diff
@@ -422,4 +417,16 @@ mod tests {
         let reader = Reader::from_path("test/test_left.vcf");
         assert!(reader.is_err());
     }
+
+    #[test]
+    fn test_bed_headered() {
+        // This file has chromosome, start, and end positions with a header line.
+        Reader::from_path("test/test_bed4.bed.gz").expect("Error opening file.");
+        // This file has chromosome and position with a header line, indexed with
+        // `tabix -b2 -e2 <file>`.
+        Reader::from_path("test/test_bed5.bed.gz").expect("Error opening file.");
+        // This is a duplicate of the above file but the index file is nonsense text.
+        Reader::from_path("test/test_bed_bad_index.bed.gz")
+            .expect_err("Invalid index file should fail.");
+    }
```

Comment:
I really like those test cases. But to avoid any future confusion, I think we should only call any file a bed file that conforms to (what pretty much amounts to) the specification of BED files here:
https://www.genome.ucsc.edu/FAQ/FAQformat.html#format1

This only applies to `test/test_bed3.bed.gz` (which I would add as another test case, here;). For `test/test_bed[45].bed.gz`, we have an unallowed type of header line as the UCSC definition linked above only allows `track` header lines. And for `test/test_bed5.bed.gz` we miss the third required column (`end`) for this file to be called a bed file. For tabix indexing, this is fine, but I would simply call them something else (e.g. `4` a `genomic_regions.txt` file and `5` a `genomic_positions.txt` file?).

And another very little change I would suggest, is to move all the test files for this reader into a subfolder in test, e.g. `test/test_tabix_reader/`. This way we don't create one huge folder of test files for all kinds of different functionality as we add tests...

Reply:
Great suggestions! I've followed them in cee3c88, with the exception of including `test_bed3.bed.gz` in these tests, since it is covered at a basic level elsewhere in this test module. I also split this function into 3 separate test functions (one for each file) so if they fail in the future it is easy to tell which one is problematic.

Reply:
Ah, OK. I simply didn't see that bed3 test in the changes...